cdc/model/schema_storage.go

```diff
@@ -146,27 +146,10 @@ func (ti *TableInfo) findHandleIndex() {
 		// pk is handle
 		return
 	}
-	handleIndexOffset := -1
-	for i, idx := range ti.Indices {
-		if !ti.IsIndexUnique(idx) {
-			continue
-		}
-		if idx.Primary {
-			handleIndexOffset = i
-			break
-		}
-		if handleIndexOffset < 0 {
-			handleIndexOffset = i
-		} else {
-			if len(ti.Indices[handleIndexOffset].Columns) > len(ti.Indices[i].Columns) ||
-				(len(ti.Indices[handleIndexOffset].Columns) == len(ti.Indices[i].Columns) &&
-					ti.Indices[handleIndexOffset].ID > ti.Indices[i].ID) {
-				handleIndexOffset = i
-			}
-		}
-	}
-	if handleIndexOffset >= 0 {
-		ti.HandleIndexID = ti.Indices[handleIndexOffset].ID
+
+	pk := ti.GetPrimaryKey()
```

Comment:
`ti.GetPrimaryKey` does not completely equal to original logic, given the scenario of no primary key
- If there is no UNIQUE INDEX on NOT NULL columns, `ti.GetPrimaryKey` returns nil, but original logic could choose one unique index
- If there exist multiple NOT NULL unique indexes, `ti.GetPrimaryKey` always return the first one, but original logic choose one with fewer columns (it has no correctness problem, but should check out whether exists compatibility issue)